packages/loader/container-loader/src/container.ts

```diff
@@ -1546,6 +1559,22 @@ export class Container
 		this._deltaManager.connect(args);
 	}
 
+	private readonly metadataUpdateHandler = (metadata?: Record<string, unknown> | undefined) => {
+		this._containerMetadata = { ...this._containerMetadata, ...metadata };
```

Comment:
This looks fishy (in terms of merging old & new).
Assuming that we get data only through joinSession, I see not much difference between first call and second call to joinSession. I'd assume that it returns full set of properties (not just update from previous state), as otherwise we will not have full info (ever) - we have no initial state.
And if it's full set, then property missing in second update should be - it's gone. But if we merge like that, it will stay around.
Let's clarify (through comments and code actions) semantics of these updates.

Reply:
The join session will always return the labels and not just the updates even if they are empty. So as far as labels is concerned, we will have the state always.  I want to go with this semantic for these updates as you previously mentioned:

**Merging properties / communicates only what changed:** If some property is missing on this payload, but existed before (in containerMetadata state below), then it did not change and stays the same. If property needs to be deleted, it's value will be set to undefined in metadata value on this event.

As then the update provider does not need to bother about all props and even container can itself add props to the metadata as the service might not know all the props, so complete overwrite semantics does not work. If someone wants to reset a prop, then they need to provide that in update otherwise it will remain same.